queries.sql

```diff
@@ -7,4 +7,71 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name != 'Gabumon';
-SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+/* Second entry*/
+
+-- Transaction 1
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+
+TABLE animals;
+
+ROLLBACK;
+
+TABLE animals;
+
+-- Transaction 2
+BEGIN;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+-- Transaction 3
+BEGIN;
+
+DELETE FROM animals;
+TABLE animals;
+
+ROLLBACK;
+
+TABLE animals;
+
+-- Transaction 4
+BEGIN;
+
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT SP1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO SP1;
+
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+
+-- Queries to answer provided questions
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, SUM(escape_attempts) FROM animals 
+GROUP BY neutered;
```

Comment:
- [ OPTIONAL ] I think you could use here Max instead of sum to answer the question Who escapes the most,since it want to ask who is the most so you can use MAX 🔥💯✅

Reply:
Thank you for the insight